OpenEdXMobile/src/main/java/org/edx/mobile/module/prefs/PrefBaseManager.kt

```diff
@@ -0,0 +1,215 @@
+     * @param defaultValue
+     * @return boolean
+     */
+    fun getBoolean(key: String, defaultValue: Boolean): Boolean =
+        sharedPreferences.getBoolean(key, defaultValue)
+
+    /**
+     * Returns long value for the given key, -1 if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return long
+     */
+    fun getLong(key: String, defaultValue: Long = -1): Long =
+        sharedPreferences.getLong(key, defaultValue)
+
+    /**
+     * Returns float value for the given key, defaultValue if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return float
+     */
+    fun getFloat(key: String, defaultValue: Float = -1f): Float =
+        sharedPreferences.getFloat(key, defaultValue)
+
+    /**
+     * Returns int value for the given key, defaultValue if no value is found.
+     *
+     * @param key,default value
+     * @return int
+     */
+    fun getInt(key: String, defaultValue: Int = -1): Int =
+        sharedPreferences.getInt(key, defaultValue)
+
+    fun removeKey(key: String) = editor.remove(key).apply()
+
+    fun clear() = editor.clear().apply()
+
+    fun migrateData(oldPrefBaseManager: PrefBaseManager) {
+        oldPrefBaseManager.sharedPreferences.all.forEach {
+            when (it.value) {
+                is Int -> put(it.key, it.value as Int)
+                is Long -> put(it.key, it.value as Long)
+                is Float -> put(it.key, it.value as Float)
+                is Boolean -> put(it.key, it.value as Boolean)
+                is String -> put(it.key, it.value as String)
+            }
+            oldPrefBaseManager.removeKey(it.key)
+        }
+    }
+
+    /**
+     * Contains preference name constants. These must be unique.
+     */
+    object Pref {
+        const val LOGIN = "pref_login"
+        const val WIFI = "pref_wifi"
+        const val VIDEOS = "pref_videos"
+        const val APP_FEATURES = "app_features"
+        const val COURSE_CALENDAR_PREF = "course_calendar_pref"
+        const val APP_INFO = "pref_app_info"
+        const val USER_PREF = "pref_user"
+        val all: Array<String>
+            get() = arrayOf(
+                LOGIN,
+                WIFI,
+                VIDEOS,
+                APP_FEATURES,
+                COURSE_CALENDAR_PREF,
+                APP_INFO,
+                USER_PREF
+            )
+
+        @JvmStatic
+        val allPreferenceFileNames: Array<String>
+            get() {
+                val preferencesFilesList = all
+                for (i in preferencesFilesList.indices) {
+                    preferencesFilesList[i] += ".xml"
+                }
+                return preferencesFilesList
+            }
+    }
+
+    object Value {
+        /*
+         * These values are used in API endpoint
+         */
+        const val BACKEND_FACEBOOK = "facebook"
+        const val BACKEND_GOOGLE = "google-oauth2"
+        const val BACKEND_MICROSOFT = "azuread-oauth2"
```

Comment:
We can remove the object Value and move them to Companion of LoginPrefs where it makes more sense.